src/constraints/field.rs

```diff
@@ -325,54 +210,40 @@ fn equal_cells<B: Backend>(
                 span,
             );
 
+            // todo: replace these front end constraints with backend constraints
```

Comment:
oh, I mean using the lower layer backend api instead (the compiler.backend.add/neg etc). But doing this would introduce repeated code that can be encapsulated the front end api, such as the `sub`.